gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -432,7 +432,7 @@ private WorkUnit getWorkUnitForTopicPartition(KafkaPartition partition, SourceSt
       } catch (StartOffsetOutOfRangeException e) {
 
         // Increment counts, which will be reported as job metrics
-        if (offsets.getStartOffset() <= offsets.getLatestOffset()) {
+        if (previousOffset <= offsets.getEarliestOffset()) {
```

Comment:
Why the change to offsets.getEarliestOffset() here? I think either leave it as `<= offsets.getLatestOffset()` or change it to `< offsets.getEarliestOffset()`